### ISSUE TITLE
Statusbar adds "^" fillchars in Ataraxis mode

### ISSUE THREAD

**Issue by Mange** [Pocco81/true-zen.nvim#100]
I'm not sure why this happens to me specifically, but [it appears to be standard behavior of Vim](https://vi.stackexchange.com/questions/15873/carets-in-status-line), so I would presume others have the same issue.

![image](https://user-images.githubusercontent.com/1599/183348019-381f75be-d579-4768-8318-89e6c97edf4f.png)

When the main window is focused, then the statusbar adds the fillchars. When focusing another "window", then that window gets it instead.

![image](https://user-images.githubusercontent.com/1599/183348119-e64c63a2-406f-464f-b54a-ed1ebbd4c122.png)

Left, Right, and Bottom windows do not get this.

According to the above linked StackExchange link, this happens when `StatusLine` and `StatusLineNC` have the same highlight.

Here's a different discussion in `nord-vim` about how they worked around it: https://github.com/arcticicestudio/nord-vim/issues/37

Vim's manual says that `fillchars` is both a global value or a window-local value. Perhaps we could override the `fillchars` to a non-breaking space and restore it when exiting the mode?

**Comment by Pocco81** [comment]
@Mange have you tried setting this in your true-zen config:

```
statusline = "%#Normal#"
```
?

**Comment by Mange** [comment]
Added this option:

![image](https://user-images.githubusercontent.com/1599/183605787-a89dbaab-ae36-411c-bd69-a9b7d659c489.png)


Still happens…

![image](https://user-images.githubusercontent.com/1599/183605997-adcc2a01-2fdb-4c20-8f19-297df199851f.png)

Trying just `:TZMinimalist` and manually splitting still shows this appearing under the focused window.

![image](https://user-images.githubusercontent.com/1599/183606343-2e00d22f-205a-4c9f-9788-6bad78417380.png)

This does not happen in my theme before or after `:TZMinimalist` since the highlights are different.

![image](https://user-images.githubusercontent.com/1599/183606617-9c97df16-d3cf-4044-981d-da9f5143aae1.png)

**Comment by loqusion** [comment]
@Mange 1488013 makes it so whatever `'statusline'` you set for minimalist is overridden when you enable lualine integration.

I've actually been experiencing issues similar to you, and I made a [fork](https://github.com/loqusion/true-zen.nvim/tree/some-fixes) of the project that fixes them. I haven't made a PR since I made some breaking changes and did quite a bit of opinionated refactoring that made it easier for me to reason about the codebase. In addition to fixing this issue, minimalist and ataraxis are (mostly) decoupled so that you can configure minimalist without those changes trickling down to ataraxis, enabling one mode disables all the other modes first to stop nesting issues, `integrations.lualine` config was removed so that the code detects lualine and hides it automatically, and the `_pos` callbacks were renamed to `_post`.

If you want to try it out to see if that fixes the `^`'s for you, be sure to change the repo name to `loqusion/true-zen.nvim` and remove the `integrations.lualine` from your true-zen config. I'm not sure if I'll make a PR yet since I don't want to intrude too much upon @Pocco81's intentions for this codebase.

If you do try it, let me know how it works out for you!

**Comment by Mange** [comment]
Still happens in your fork, sadly. :-(

![image](https://user-images.githubusercontent.com/1599/184423750-8a33b2b0-e796-4b0f-8e38-7471b34c87f8.png)

(And since Lualine integration is removed Lualine stays visible.)

**Comment by loqusion** [comment]
I just cloned your dotfiles and here's what I see:

![image](https://user-images.githubusercontent.com/38332081/184444061-5c284bf9-1e79-4a31-aad5-2d2106b96efe.png)

Perhaps you forgot to add `branch = 'some-fixes'`? I know, it's kinda weird :P

**Comment by Mange** [comment]
Ha ha, yeah. I must've forgotten that change. I can confirm I get the correct results now in your fork. :facepalm: 

Thank you!

**Comment by oblitum** [comment]
It's my second attempt trying this plugin after a long time (I've been using folke/zen-mode.nvim), and this issue comes up right away. @loqusion's fork indeed fixes the problem.

**Comment by kanielrkirby** [comment]
> @Mange [1488013](https://github.com/pocco81/true-zen.nvim/commit/1488013a8e128e02ee1e20f4cb00c2dd0e595319) makes it so whatever `'statusline'` you set for minimalist is overridden when you enable lualine integration.
> 
> I've actually been experiencing issues similar to you, and I made a [fork](https://github.com/loqusion/true-zen.nvim/tree/some-fixes) of the project that fixes them. I haven't made a PR since I made some breaking changes and did quite a bit of opinionated refactoring that made it easier for me to reason about the codebase. In addition to fixing this issue, minimalist and ataraxis are (mostly) decoupled so that you can configure minimalist without those changes trickling down to ataraxis, enabling one mode disables all the other modes first to stop nesting issues, `integrations.lualine` config was removed so that the code detects lualine and hides it automatically, and the `_pos` callbacks were renamed to `_post`.
> 
> If you want to try it out to see if that fixes the `^`'s for you, be sure to change the repo name to `loqusion/true-zen.nvim`, add `branch = 'some-fixes'` to your packer config, and remove the `integrations.lualine` from your true-zen config. I'm not sure if I'll make a PR yet since I don't want to intrude too much upon @pocco81's intentions for this codebase.
> 
> If you do try it, let me know how it works out for you!

Nearly 2 years later, and I ran into this issue. Searched through the plugin's issues and found you, don't know how many people actually ran into this at all, but we appreciate you lol. Really unfortunate problem with this plugin, as I run into lag issues with folke's zen plugin (love folke, I probably just did something wrong with my config, but this plugin has everything I need and I've already spent a couple too many hours configuring my writing environment for the day haha).

TL;DR, thanks man.
